### PR TITLE
Add API endpoint to copy documents from a workspace into GEVER.

### PR DESCRIPTION
**Pull request by njohner**
We add two endpoints:

- Endpoint to list documents in a linked workspace:
  - We handle batching

- Endpoint to copy documents back from a linked workspace into GEVER. 
  - We take care to also handle eml and msg mails

Moreover we had to correct several bugs:
- handle batching in endpoint listing linked workspaces
- allow E-mails in `workspace_folder`s
- handle E-mails in the `copy-mail-to-workspace`
- avoid posting other file fields in the `copy-mail-to-workspace` endpoint

## Checkliste

- [x] Gibt es neue Funktionalität mit einem `Dokument`? Funktioniert das auch mit einem `Mail`?
- [x] Sind UpgradeSteps `deferrable`, oder können gewisse Schritte des Upgrades konditional ausgeführt werden? Upgrades are not deferrable
- [x] Changelog-Eintrag vorhanden/nötig?
- [x] Aktualisierung Dokumentation vorhanden/nötig?

For https://github.com/4teamwork/opengever.core/issues/6157
Fixes https://sentry.4teamwork.ch/sentry/onegov-gever/issues/62613/

### CODE REVIEW

File: opengever/core/upgrades/20200304085353_allow_emails_in_workspace_folders/types/opengever.workspace.folder.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<object name="opengever.meeting.proposal">
```
Comment (lukasgraf) [inline opengever/core/upgrades/20200304085353_allow_emails_in_workspace_folders/types/opengever.workspace.folder.xml]: Huh? This doesn't seem right.

File: opengever/workspaceclient/linked_workspaces.py [hunk 2]
```diff
@@ -113,6 +128,38 @@ def copy_document_to_workspace(self, document, workspace_uid):
                                       size, content_type, filename,
                                       **self._tus_document_repr(document_repr))
 
+    def copy_document_from_workspace(self, document_uid):
+        """Will upload a copy of a document to a linked workspace.
```
Comment (lukasgraf) [inline opengever/workspaceclient/linked_workspaces.py]: Lies 😉

File: opengever/workspaceclient/linked_workspaces.py [hunk 3]
```diff
@@ -113,6 +128,38 @@ def copy_document_to_workspace(self, document, workspace_uid):
...
+        """
+        document_url = self.client.lookup_url_by_uid(document_uid)
+        document_repr = self.client.get(url_or_path=document_url)
+
+        if document_repr.get('archival_file'):
+            data = self.client.request.get(document_repr['archival_file']['download'])
+            document_repr['file']['data'] = data.content
```
Comment (lukasgraf) [inline opengever/workspaceclient/linked_workspaces.py]: That should be save to `document_repr['archival_file']['data']`, no?

File: opengever/workspaceclient/linked_workspaces.py [hunk 4]
```diff
@@ -40,6 +42,19 @@ def list_cache_key(linked_workspaces_instance, **kwargs):
     return cache_key
 
 
+class ProxyPost(GeverFolderPost):
```
Comment (lukasgraf) [inline opengever/workspaceclient/linked_workspaces.py]: It's not quite clear to me what this is / does.

File: docs/public/dev-manual/api/linked_workspaces.rst [hunk 5]
```diff
@@ -105,3 +105,104 @@ Achtung: Die Kopie wird nicht mit dem originalen Dokument verknüpft. Es handelt
       "title": "Ein Dokument",
        "...": "..."
     }
+
+Dokumente in einem Verknüpftem Teamraum auflisten
```
Comment (lukasgraf) [inline docs/public/dev-manual/api/linked_workspaces.rst]: verknüpften

File: docs/public/dev-manual/api/linked_workspaces.rst [hunk 6]
```diff
@@ -105,3 +105,104 @@ Achtung: Die Kopie wird nicht mit dem originalen Dokument verknüpft. Es handelt
...
+-------------------------------------------------
+
+Über den Endpoint `@list-documents-in-linked-workspace` werden die Dokumente in einem verlinktem Teamraum aufgelistet. Der Endpoint benötigt als zusätzlichen Pfad Parameter die UID des Teamraums, i.e. `@list-documents-in-linked-workspace/workspace_uid`. Dieser Endpoint unterstützt Batching.
```
Comment (lukasgraf) [inline docs/public/dev-manual/api/linked_workspaces.rst]: RST wants double backticks for marking up literals.
`i.e.` is `d.h.` in German 😉 (and `e.g.` is `z.B.`, which would fit better here IMHO).

File: docs/public/dev-manual/api/linked_workspaces.rst [hunk 7]
```diff
@@ -105,3 +105,104 @@ Achtung: Die Kopie wird nicht mit dem originalen Dokument verknüpft. Es handelt
...
+              "filename": "Test email.eml",
+              "review_state": "mail-state-active",
+              "title": "Test email"
+          }
+        ],
+      "items_total": 2
+    }
+
+
+**Beispiel-Response**:
```
Comment (lukasgraf) [inline docs/public/dev-manual/api/linked_workspaces.rst]: This seems extraneous - left in by accident?

File: opengever/workspaceclient/linked_workspaces.py (hunk 2 above)
Comment (njohner) [reply opengever/workspaceclient/linked_workspaces.py]: Looks like a rebase fail, it's not in the final code 😓

File: opengever/workspaceclient/linked_workspaces.py (hunk 4 above)
Comment (njohner) [reply opengever/workspaceclient/linked_workspaces.py]: I added docstrings

File: docs/public/dev-manual/api/linked_workspaces.rst (hunk 6 above)
Comment (njohner) [reply docs/public/dev-manual/api/linked_workspaces.rst]: everybody knows latin :-D

File: docs/public/dev-manual/api/linked_workspaces.rst (hunk 7 above)
Comment (njohner) [reply docs/public/dev-manual/api/linked_workspaces.rst]: yep sorry

**Comment by njohner** [comment]
@lukasgraf I updated the PR. Corrected everything you suggested and added docstrings to explain what the `ProxyPost` class is used for.

**Pull request merged**